### PR TITLE
Updated MLJ interface for feature importances per #166

### PR DESCRIPTION
**Pull request by john-waczak**

### CODE REVIEW

**Comment by ablaom** [comment]
@OkonSamuel

File: src/MLJ.jl [hunk 1]
```diff
@@ -45,7 +46,8 @@ function MMI.update(model::EvoTypes, verbosity::Integer, fitresult, cache, A, y)
     grow_evotree!(fitresult, cache)
   end
 
-  report = (feature_importances=importance(fitresult, A.names),)
+  # report = (feature_importances=importance(fitresult, A.names),)
+  report = ()
```
Comment (jeremiedb) [inline src/MLJ.jl:50]: Shouldn't it return the feature names, same as in the above `fit` function? 
`report = (features=A.names,)`

File: src/MLJ.jl [hunk 2]
```diff
@@ -5,7 +5,8 @@ function MMI.fit(model::EvoTypes, verbosity::Int, A, y)
     fitresult, cache = init_evotree(model, A.matrix, y)
   end
   grow_evotree!(fitresult, cache)
-  report = (feature_importances=importance(fitresult, A.names),)
+#  report = (feature_importances=importance(fitresult, A.names),)
```
Comment (jeremiedb) [inline src/MLJ.jl:8]: I'd remove the commented out line.

File: src/MLJ.jl [hunk 3]
```diff
@@ -45,7 +46,8 @@ function MMI.update(model::EvoTypes, verbosity::Integer, fitresult, cache, A, y)
     grow_evotree!(fitresult, cache)
   end
 
-  report = (feature_importances=importance(fitresult, A.names),)
+  # report = (feature_importances=importance(fitresult, A.names),)
```
Comment (jeremiedb) [inline src/MLJ.jl:49]: Remove commentted out line.

File: src/MLJ.jl [hunk 4]
```diff
@@ -70,6 +72,22 @@ function predict(::EvoTreeGaussian, fitresult, A)
...
+
+# # Feature Importances
+MMI.reports_feature_importances(::Type{<:EvoTreeClassifier}) = true
+MMI.reports_feature_importances(::Type{<:EvoTreeRegressor}) = true
+MMI.reports_feature_importances(::Type{<:EvoTreeCount}) = true
+MMI.reports_feature_importances(::Type{<:EvoTreeGaussian}) = true
+
+function MMI.feature_importances(m::Union{EvoTreeClassifier, EvoTreeRegressor, EvoTreeCount, EvoTreeGaussian}, fitresult, report)
+    fi_pairs = importance(fitresult, report.features)
+    sort!(fi_pairs, by= x->-x[2])
```
Comment (jeremiedb) [inline src/MLJ.jl:84]: Can be removed since `importance` already sort the feature in decreasing order.

File: src/MLJ.jl [hunk 5]
```diff
@@ -70,6 +72,22 @@ function predict(::EvoTreeGaussian, fitresult, A)
   return [Distributions.Normal(pred[i, 1], pred[i, 2]) for i = 1:size(pred, 1)]
 end
 
+
+# # Feature Importances
+MMI.reports_feature_importances(::Type{<:EvoTreeClassifier}) = true
+MMI.reports_feature_importances(::Type{<:EvoTreeRegressor}) = true
+MMI.reports_feature_importances(::Type{<:EvoTreeCount}) = true
+MMI.reports_feature_importances(::Type{<:EvoTreeGaussian}) = true
```
Comment (jeremiedb) [inline src/MLJ.jl:80]: Can be simplified by using `EvoTypes`: 
```julia
MMI.reports_feature_importances(::Type{<:EvoTypes}) = true
```

**Review by jeremiedb — CHANGES_REQUESTED** [review]
Thanks for bringin this PR! 
I've made some comments regarding suggested edits.

**Comment by john-waczak** [comment]
@jeremiedb Comments should be gone now, update method is fixed, and I removed the repeated calls to `reports_feature_importances` per your suggestion. Please let me know if there's anything else I need to  tweak!

**Comment by jeremiedb** [comment]
@john-waczak Thans for the update. Just for clarification, although I understand that MLJ provides a seperate `features importance` extractor function, was there actually a concern for having the feature importance calculated within the report? Is the reason to save compute time while iterating over a model?

**Comment by OkonSamuel** [comment]
> @john-waczak Thans for the update. Just for clarification, although I understand that MLJ provides a seperate `features importance` extractor function, was there actually a concern for having the feature importance calculated within the report? Is the reason to save compute time while iterating over a model?

@jeremiedb MLJ intends to use that API to build other functionality such are recursive feature elimination etc. But it is still going to be accessible by the user when calling `feature_importances(mach::Machine)`. There is nothing stopping you from leaving feature importances in your report then just returning that again in when one calls `feature_importances(model, fitresult, report)`. 
Again, pushing the feature importances calculations to the `feature_importances` method can help to save some computation time. It is only computed when the user needs it.

**Comment by john-waczak** [comment]
@jeremiedb @OkonSamuel I can add them back to the report if you prefer. I just figured with the newer extraction method it made sense to remove the duplicate in the report.

**Review by jeremiedb — APPROVED** [review]

**Comment by jeremiedb** [comment]
I just wanted to make sure I understood the original motivation for the changes. Sounds good to me, thanks for @john-waczak and @OkonSamuel for the clarifications.

**Comment by ablaom** [comment]
Just wanted to be sure that the importances returned has the correct form. Is an `AbstractVector` ([required](https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use/#Feature-importances)) or `Tuple` of pairs returned? In any case, I suggest explicitly testing the importances have the required form.

**Comment by ablaom** [comment]
Okay, I didn't update my page and now see this is merged with a review from Samuel, so all should be good.

**Pull request merged**